pkg/secretless/plugin/connector/http/http.go

```diff
@@ -0,0 +1,24 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+)
+
+// Plugin is the main interface that HTTP plugins need to implement
+// to be loaded by our codebase.
+type Plugin interface {
+	// PluginInfo contains a key-value map of informational fields
+	// about the plugin.
+	PluginInfo() map[string]string
+
+	// NewConnector creates a new Connector based on the ConnectorResources
+	// passed into it.
+	NewConnector(connector.Resources) Connector
+}
+
+// Connector is the function that will be invoked when a matching
+// request comes in. It uses both the request object and the credentials
+// map to authenticate the client.
+type Connector func(request *http.Request, credentials map[string][]byte) error
```

Comment:
I'm thinking we ought to create a meaningfully named type for `map[string][]byte`.  Something like:
```go
type SecretsByName map[string][]byte
// OR
type RawSecrets map[string][]byte
```
This won't affect usage at all.  See:

https://play.golang.org/p/0AdmgxVweOO